client/header/activity-panel/tabs/index.js

```diff
@@ -0,0 +1,67 @@
+import { NavigableMenu } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+import { Tab } from '../tab';
+import { recordEvent } from 'lib/tracks';
+
+export const Tabs = ( {
+	tabs,
+	onTabClick,
+	selectedTab: selectedTabName,
+	tabOpen = false,
+} ) => {
+	const [ tabState, setTabState ] = useState( {
+		tabOpen,
+		currentTab: selectedTabName,
+	} );
+
+	// Keep state synced with props
+	useEffect( () => {
+		setTabState( {
+			tabOpen,
+			currentTab: selectedTabName,
+		} );
+	}, [ tabOpen, selectedTabName ] );
+
+	return (
+		<NavigableMenu
+			role="tablist"
+			orientation="horizontal"
+			className="woocommerce-layout__activity-panel-tabs"
+		>
+			{ tabs &&
+				tabs.map( ( tab, i ) => (
+					<Tab
+						key={ i }
+						index={ i }
+						isPanelOpen={ tabState.tabOpen }
```

Comment:
Any thoughts on destructing `tabState` into the needed attributes used here - or do you like having `tabState.` and `tab.` to kind of keep the logic easier to follow about what attributes are being compared?

Reply:
I think you could go either way with this, but I did refactor it to destructure the state. The attributes are a liittle more confusing but its also more succinct, so I'm happy with that change :+1: